controller/src/beerocks/bml/internal/bml_internal.cpp

```diff
@@ -1401,6 +1433,89 @@ int bml_internal::get_dcs_continuous_scan_params(const std::string &mac, int *ou
     return (iRet);
 }
 
+int bml_internal::get_dcs_scan_results(const std::string &mac, BML_NEIGHBOR_AP **output_results,
+                                       unsigned int *output_results_size,
+                                       const unsigned int max_results_size,
+                                       uint8_t *output_result_status, bool is_single_scan)
+{
+    if (m_sockPlatform == nullptr && !connect_to_platform()) {
+        return (-BML_RET_CONNECT_FAIL);
+    }
+
+    // Initialize the promise for receiving the response
+    beerocks::promise<int> prmChannelScanResultsGet;
+    m_prmChannelScanResultsGet    = &prmChannelScanResultsGet;
+    int iOpTimeout                = DELAYED_RESPONSE_TIMEOUT; // Default timeout
+    auto scan_results             = std::list<beerocks_message::sChannelScanResults>();
+    uint32_t scan_results_size    = 0;
+    uint32_t scan_results_maxsize = uint32_t(max_results_size);
+
+    m_scan_results         = &scan_results;
+    m_scan_results_size    = &scan_results_size;
```

Comment:
I don't understand why we need both `m_scan_results_size` and `m_scan_results_maxsize`